x/concentrated-liquidity/model/pool.go

```diff
@@ -118,9 +119,9 @@ func (p Pool) SpotPrice(ctx sdk.Context, baseAssetDenom string, quoteAssetDenom
 	}
 
 	if baseAssetDenom == p.Token0 {
-		return p.CurrentSqrtPrice.Power(2), nil
+		return p.CurrentSqrtPrice.PowerInteger(2).SDKDec(), nil
 	}
-	return sdk.NewDec(1).Quo(p.CurrentSqrtPrice.Power(2)), nil
+	return osmomath.OneDec().Quo(p.CurrentSqrtPrice.PowerInteger(2)).SDKDec(), nil
```

Comment:
note: these truncate at the precision end which should be fine since this is a spot price query